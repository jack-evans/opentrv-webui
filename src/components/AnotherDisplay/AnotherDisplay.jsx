import React, { Component } from 'react'

class AnotherDisplay extends Component {
  render () {
    return (
      <div>
        <p>
          I'm another display panel
        </p>
      </div>
    )
  }
}

export default AnotherDisplay
