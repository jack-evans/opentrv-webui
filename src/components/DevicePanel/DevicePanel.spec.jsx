import React from 'react'
import AnotherDisplay from './DevicePanel'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('AnotherDisplay.jsx', () => {
  it('renders without crashing', () => {
    const match = {
      params: {
        id: '1234'
      }
    }

    expect(shallow(<AnotherDisplay match={match} />)).toMatchSnapshot()
  })
})
