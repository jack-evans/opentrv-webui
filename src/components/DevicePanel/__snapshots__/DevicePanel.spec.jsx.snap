// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnotherDisplay.jsx renders wthout crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DevicePanel
    match={
      Object {
        "params": Object {
          "id": "1234",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="DevicePanel__header-container"
      >
        <a
          className="DevicePanel__header-back-button"
          href="/"
        >
          <Icon
            description="Provide a description that will be used as the title"
            fillRule="evenodd"
            name="arrow--left"
            role="img"
          />
          <h3>
            Device Overview
          </h3>
        </a>
      </div>,
      "className": "DevicePanel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="DevicePanel__header-back-button"
          href="/"
        >
          <Icon
            description="Provide a description that will be used as the title"
            fillRule="evenodd"
            name="arrow--left"
            role="img"
          />
          <h3>
            Device Overview
          </h3>
        </a>,
        "className": "DevicePanel__header-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              description="Provide a description that will be used as the title"
              fillRule="evenodd"
              name="arrow--left"
              role="img"
            />,
            <h3>
              Device Overview
            </h3>,
          ],
          "className": "DevicePanel__header-back-button",
          "href": "/",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "description": "Provide a description that will be used as the title",
              "fillRule": "evenodd",
              "name": "arrow--left",
              "role": "img",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Device Overview",
            },
            "ref": null,
            "rendered": "Device Overview",
            "type": "h3",
          },
        ],
        "type": "a",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="DevicePanel__header-container"
        >
          <a
            className="DevicePanel__header-back-button"
            href="/"
          >
            <Icon
              description="Provide a description that will be used as the title"
              fillRule="evenodd"
              name="arrow--left"
              role="img"
            />
            <h3>
              Device Overview
            </h3>
          </a>
        </div>,
        "className": "DevicePanel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="DevicePanel__header-back-button"
            href="/"
          >
            <Icon
              description="Provide a description that will be used as the title"
              fillRule="evenodd"
              name="arrow--left"
              role="img"
            />
            <h3>
              Device Overview
            </h3>
          </a>,
          "className": "DevicePanel__header-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                description="Provide a description that will be used as the title"
                fillRule="evenodd"
                name="arrow--left"
                role="img"
              />,
              <h3>
                Device Overview
              </h3>,
            ],
            "className": "DevicePanel__header-back-button",
            "href": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "description": "Provide a description that will be used as the title",
                "fillRule": "evenodd",
                "name": "arrow--left",
                "role": "img",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Device Overview",
              },
              "ref": null,
              "rendered": "Device Overview",
              "type": "h3",
            },
          ],
          "type": "a",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
